Microsoft.Toolkit.Uwp.UI/Extensions/Tree/LogicalTree.cs

```diff
@@ -595,15 +595,41 @@ public static IEnumerable<FrameworkElement> FindParents(this FrameworkElement el
         /// If the key is not found in the current element's resources, the logical tree is then
         /// searched element-by-element to look for the resource in each element's resources.
         /// If none of the elements contain the resource, the Application's resources are then searched.
-        /// <para>See: <seealso href="https://docs.microsoft.com/dotnet/api/system.windows.frameworkelement.tryfindresource"/></para>
-        /// <para>And also: <seealso href="https://docs.microsoft.com/dotnet/desktop-wpf/fundamentals/xaml-resources-define#static-resource-lookup-behavior"/></para>
+        /// <para>See: <seealso href="https://docs.microsoft.com/dotnet/api/system.windows.frameworkelement.tryfindresource"/>.</para>
+        /// <para>And also: <seealso href="https://docs.microsoft.com/dotnet/desktop-wpf/fundamentals/xaml-resources-define#static-resource-lookup-behavior"/>.</para>
         /// </summary>
         /// <param name="element">The <see cref="FrameworkElement"/> to start searching for the target resource.</param>
         /// <param name="resourceKey">The resource key to search for.</param>
-        /// <returns>The requested resource, or <see langword="null"/>.</returns>
-        public static object? TryFindResource(this FrameworkElement element, object resourceKey)
+        /// <returns>The requested resource.</returns>
+        /// <exception cref="Exception">Thrown when no resource is found with the specified key.</exception>
+        public static object FindResource(this FrameworkElement element, object resourceKey)
         {
-            object? value = null;
+            if (TryFindResource(element, resourceKey, out object? value))
+            {
+                return value!;
+            }
+
+            static object Throw() => throw new Exception("No resource was found with the specified key");
```

Comment:
Would a `KeyNotFoundException` or something be better?

Reply:
Oh, definitely! Fixed in 55a0583bde203377953579a1cc270753d93641ad, good idea! 😄
Also added the resource key in the exception message while I was at it.